datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -1991,6 +1991,56 @@ pub fn array_intersect(args: &[ArrayRef]) -> Result<ArrayRef> {
     }
 }
 
+/// array_distinct SQL function
+/// example: from list [1, 3, 2, 3, 1, 2, 4] to [1, 2, 3, 4]
+pub fn array_distinct(args: &[ArrayRef]) -> Result<ArrayRef> {
+    assert_eq!(args.len(), 1);
+
+    // handle null
+    if args[0].data_type() == &DataType::Null {
+        return Ok(args[0].clone());
+    }
+
+    let array = as_list_array(&args[0])?;
```

Comment:
Could you extend it to `LargeList` too?

Reply:
something like this `arrow_cast(array, 'LargeList(Int64)')`?

Reply:
get it! ,thanks @jayzhan211 @Weijun-H